File: tests/Generator/resource/swift/complex/GenericPropertyType.swift
// Represents a generic value which can be replaced with a concrete type
class GenericPropertyType: PropertyType {
    var name: String?
    var _type: String? = "generic"

    enum CodingKeys: String, CodingKey {
        case name = "name"
        case _type = "type"
    }
}

File: tests/Generator/resource/python/complex/generic_property_type.py
from pydantic import BaseModel, Field, GetCoreSchemaHandler, Tag
from pydantic_core import CoreSchema, core_schema
from typing import Any, Dict, Generic, List, Optional, TypeVar, Annotated, Union, Literal
from .property_type import PropertyType


# Represents a generic value which can be replaced with a concrete type
class GenericPropertyType(PropertyType):
    type: Literal["generic"] = Field(alias="type")
    name: Optional[str] = Field(default=None, alias="name")
    pass


